src/features/journeys/components/JourneyInstanceCloseButton.tsx

```diff
@@ -74,10 +52,17 @@ const JourneyInstanceCloseButton: React.FunctionComponent<{
       <ZUIDialog onClose={closeAndClear} open={showDialog}>
         <Box data-testid="JourneyInstanceCloseButton-outcomeDialog">
           <form
-            onSubmit={(e) => {
+            onSubmit={async (e) => {
               e.preventDefault();
               e.stopPropagation();
-              onSubmit();
+              setIsLoading(true);
+              await closeJourneyInstance({
+                closed: dayjs().toJSON(),
+                outcome: outcomeNote,
+                tags: internalTags,
+              });
+              setIsLoading(false);
```

Comment:
Nice! This will probably find it's way into some nice utility or hook at some point. 💯